src/Microsoft.OData.Core/ODataBatchWriter.cs

```diff
@@ -313,81 +313,34 @@ public Task<ODataBatchOperationResponseMessage> CreateOperationResponseMessageAs
 #endif
 
         /// <summary>Flushes the write buffer to the underlying stream.</summary>
-        public void Flush()
-        {
-            this.VerifyCanFlush(true);
-
-            // make sure we switch to state FatalExceptionThrown if an exception is thrown during flushing.
-            try
-            {
-                this.rawOutputContext.Flush();
-            }
-            catch
-            {
-                this.SetState(BatchWriterState.Error);
-                throw;
-            }
-        }
+        public abstract void Flush();
 
 #if ODATALIB_ASYNC
         /// <summary>Flushes the write buffer to the underlying stream asynchronously.</summary>
         /// <returns>A task instance that represents the asynchronous operation.</returns>
-        public Task FlushAsync()
-        {
-            this.VerifyCanFlush(false);
-
-            // make sure we switch to state FatalExceptionThrown if an exception is thrown during flushing.
-            return this.rawOutputContext.FlushAsync().FollowOnFaultWith(t => this.SetState(BatchWriterState.Error));
-        }
+        public abstract Task FlushAsync();
 #endif
 
         /// <summary>
         /// This method is called to notify that the content stream for a batch operation has been requested.
         /// </summary>
-        void IODataBatchOperationListener.BatchOperationContentStreamRequested()
-        {
-            // Write any pending data and flush the batch writer to the async buffered stream
-            this.StartBatchOperationContent();

```

Comment:
As above; you should be able to keep the implementation in the base class, and remove the implementations in the derived classes, by referencing the outputContext rather than the rawOutputContext.